zb-classifier.js

```diff
@@ -348,14 +364,33 @@ class ZigbeeClassifier {
 
   initOnOffSwitch(node, genOnOffEndpoint) {
     node.type = Constants.THING_TYPE_ON_OFF_SWITCH;
-
     this.addOnProperty(node, genOnOffEndpoint);
   }
 
   initMultiLevelSwitch(node, genLevelCtrlEndpoint) {
-    node.type = Constants.THING_TYPE_MULTI_LEVEL_SWITCH;
+    let lightLinkEndpoint =
+      node.findZhaEndpointWithInputClusterIdHex(CLUSTER_ID_LIGHTLINK_HEX);
+    if (DEBUG) {
+      console.log('     lightLinkEndpoint =', lightLinkEndpoint);
+    }
+    let colorSupported = false;
+    if (lightLinkEndpoint) {
+      if ((node.colorCapabilities & 3) != 0) {
+        // Hue and Saturation are supported
+        colorSupported = true;
+        node.type = Constants.THING_TYPE_ON_OFF_COLOR_LIGHT;
```

Comment:
@hobinjk convinced me that we should use onOffColorLight and that dimmableColorLight doesn't really make sense (because it give the user 2 different ways to set the level - one through the color picker and one through the level). I tried it originally using dimmableColorLight and changed it back to onOffColorLight after playing with the UI a bit.

Reply:
I get that the brightness/level is just a component of the color, but as a normal user, it makes more sense to me to set a color I like (white), and be able to dim that independently (as with a physical dimmer switch), rather than picking a yellowish tone that _looks like_ a dim soft white.

Any idea what the Hue app lets users do? If it typically just allows color adjustment, maybe we stick with that. With the TP-Link bulbs, the app allows both color and level adjustment. It makes sense to me to stick with whatever the manufacturer typically provides, as that's what the user probably expects.

Reply:
The color picker also shows hue, saturation and value (in addition to the RGB values). So you can use the value slider to adjust the level without changing the color. Personally, I could go either way, I decided to go for consistency with the hue plugin for the time being,

Reply:
That's fine for now. This probably warrants further discussion in the future.

Reply:
The way the Hue app does this is by constraining the color picker to have brightness = 1 (there is no brightness slider present), then having the ability to change the brightness by dragging left and right on a separate screen. Since we aren't able to easily make the color picker not have a brightness slider we should just use the color picker as a one stop shop for determining the state of the light instead of having multiple ways to do the same thing. I can copy-paste the in-depth explanation from the last conversation I had on this topic if you want.